test/common/operation.go

```diff
@@ -41,225 +41,82 @@ func (client *Client) LabelNamespace(labels map[string]string) error {
-	subscriptionMeta := base.MetaEventing(name, namespace, "Subscription")
-	if err := base.WaitForResourceReady(client.Dynamic, subscriptionMeta); err != nil {
-		return err
-	}
-	return nil
-}
-
-// WaitForSubscriptionsReady waits until all subscriptions in the namespace are Ready.
-func (client *Client) WaitForSubscriptionsReady() error {
-	namespace := client.Namespace
-	subscriptions, err := client.Eventing.EventingV1alpha1().Subscriptions(namespace).List(metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
-	for _, subscription := range subscriptions.Items {
-		if err := client.WaitForSubscriptionReady(subscription.Name); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-// WaitForCronJobSourceReady waits until the cronjobsource is Ready.
-func (client *Client) WaitForCronJobSourceReady(name string) error {
-	namespace := client.Namespace
-	cronJobSourceMeta := base.MetaSource(name, namespace, "CronJobSource")
-	if err := base.WaitForResourceReady(client.Dynamic, cronJobSourceMeta); err != nil {
-		return err
-	}
-	return nil
-}
-
-// WaitForCronJobSourcesReady waits until all cronjobsources in the namespace are Ready.
-func (client *Client) WaitForCronJobSourcesReady() error {
-	namespace := client.Namespace
-	cronJobSources, err := client.Eventing.SourcesV1alpha1().CronJobSources(namespace).List(metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
-	for _, cronJobSource := range cronJobSources.Items {
-		if err := client.WaitForCronJobSourceReady(cronJobSource.Name); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
-// WaitForContainerSourceReady waits until the containersource is Ready.
-func (client *Client) WaitForContainerSourceReady(name string) error {
-	namespace := client.Namespace
-	containerSourceMeta := base.MetaSource(name, namespace, "ContainerSource")
-	if err := base.WaitForResourceReady(client.Dynamic, containerSourceMeta); err != nil {
-		return err
-	}
-	return nil
-}
-
-// WaitForContainerSourcesReady waits until all containersources in the namespace are Ready.
-func (client *Client) WaitForContainerSourcesReady() error {
-	namespace := client.Namespace
-	containerSources, err := client.Eventing.SourcesV1alpha1().ContainerSources(namespace).List(metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
-	for _, containerSource := range containerSources.Items {
-		if err := client.WaitForContainerSourceReady(containerSource.Name); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
 // WaitForAllTestResourcesReady waits until all test resources in the namespace are Ready.
-// Currently the test resources include Pod, Channel, Subscription, Broker and Trigger.
-// If there are new resources, this function needs to be changed.
+// If there is a new resource, its TypeMeta needs to be added into the list.
+// TODO(Fredy-Z): make this function more generic by only checking existed resources in the current namespace.
 func (client *Client) WaitForAllTestResourcesReady() error {
```

Comment:
Even if the test is not for cronJobSources, this will check for readiness of cronJobSources?
Might be useful to create a function that passes the types to check?
I guess is whatever you find easier to use

Reply:
For now yes. But since every test runs in its own namespace, it does not impact much waiting for non-used resources as the returned resource list would be empty.
As a future enhancement, I will change it to only wait for created resources.
`WaitForAllTestResourcesReady` is useful as it makes the test cases conciser and easier-to-read.